tdrs-backend/tdpservice/reports/views.py

```diff
@@ -17,15 +18,43 @@
 logger = logging.getLogger()
 
 
+class ReportFileFilter(filters.FilterSet):
+    """Filters that can be applied to GET requests as query parameters."""
+
+    # Override the generated definition for the STT field so we can require it.
+    stt = filters.NumberFilter(field_name='stt_id', required=True)
+
+    class Meta:
+        """Class metadata linking to the ReportFile and fields accepted."""
+
+        model = ReportFile
+        fields = ['stt', 'quarter', 'year']
+
+
 class ReportFileViewSet(ModelViewSet):
     """Report file views."""
 
     http_method_names = ['get', 'post', 'head']
+    filterset_class = ReportFileFilter
     parser_classes = [MultiPartParser]
     permission_classes = [ReportFilePermissions]
     serializer_class = ReportFileSerializer
+
+    # TODO: Handle versioning in queryset
+    # Ref: https://github.com/raft-tech/TANF-app/issues/1007
     queryset = ReportFile.objects.all()
 
+    # NOTE: This is a temporary hack to make sure the latest version of the file
+    # is the one presented in the UI. Once we implement the above linked issue
+    # we will be able to appropriately refer to the latest versions only.
+    ordering = ['-version']
+
+    def filter_queryset(self, queryset):
+        """Only apply filters to the list action."""
+        if self.action != 'list':
```

Comment:
The filters are only relevant to the "list" view (`GET /v1/reports`) so we disable them for other actions within this viewset otherwise an error will be raised from the STT filter being required. All other actions operate on a single object (POST new report, GET a single report by ID, Download a report by ID)